core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/ResourceConverter.java

```diff
@@ -124,4 +127,32 @@ private void expandVersions(List<Resource> results) {
 		}
 	}
 	
+	private void expandCommits(List<Resource> results) {
+		if (expand().containsKey(TerminologyResource.Expand.COMMITS)) {
+			Options expandOptions = expand().getOptions(TerminologyResource.Expand.COMMITS);
+			// commit searches must be performed individually on each resource to provide correct results
+			var commitSearchRequests = results.stream()
+				.filter(TerminologyResource.class::isInstance)
+				.map(TerminologyResource.class::cast)
+				.map(res -> {
+					return RepositoryRequests.commitInfos().prepareSearchCommitInfo()
+						.filterByBranch(res.getBranchPath())
+						.setLimit(getLimit(expandOptions))
+						.setFields(expandOptions.containsKey("fields") ? expandOptions.getList("fields", String.class) : CommitInfo.Fields.DEAFULT_FIELD_SELECTION)
+						.sortBy(expandOptions.containsKey("sort") ? expandOptions.getString("sort") : null)
+						.setLocales(locales())
+						.build(res.getToolingId())
+						.execute(context().service(IEventBus.class))
```

Comment:
Do we want to run all commit search requests in parallel? Maybe a `filterByBranches` method could be added later, accepting an `Iterable<String>` of branch paths, so we end up with a single request per tooling.

Is the `commits()` expansion expected to be used in single terminology resource ("get") scenarios right now?

Reply:
It is expected to be used when you perform a search request or a single get by ID request. Both use cases should be covered.

Reply:
To properly expand commits with sorting options (like `timestamp:desc`) we need to expand them parallel for each resource.